### PR TITLE
Feature/mc 15 vue3 migration

### PR DESCRIPTION
**Pull request by mandziukGIT**

### CODE REVIEW

File: src/components/movies/MovieCard.vue [hunk 1]
```diff
@@ -16,28 +16,19 @@
...
-    movieLength() {
-      return this.movie && formatMovieLength(this.movie.length);
-    },
-    moviePoster() {
-      return this.movie?.poster_url;
-    },
-    movieCategory() {
-      return this.movie?.genre?.name;
-    },
+  props: ['movie'],
```
Comment (szymonlicau) [inline src/components/movies/MovieCard.vue]: You can have the props done in the same way here too (with the `type` and `default` options) 💡

File: src/components/movies/MovieCard.vue [hunk 2]
```diff
@@ -16,28 +16,19 @@
...
+  setup(props) {
+    const { movie } = toRefs(props);
```
Comment (szymonlicau) [inline src/components/movies/MovieCard.vue]: I'm not sure but I don't think we need to make it a ref 🤔
Any particular reason you made this one a ref?

File: src/components/register/FinalStep.vue [hunk 3]
```diff
@@ -1,137 +1,141 @@
...
+
 export default {
-    components: {
-        FormCard,
+  setup() {
+    const birthDate = ref('');
+    const { isBirthDateError, isFullAgeError, validateBirthDate } =
+      useValidate(birthDate);
+    return { isBirthDateError, isFullAgeError, validateBirthDate, birthDate };
+  },
```
Comment (szymonlicau) [inline src/components/register/FinalStep.vue]: Why not go all the way and move everything into the setup? 🚀

**Review by szymonlicau — CHANGES_REQUESTED** [review]

File: src/components/register/FinalStep.vue (hunk 3 above)
Comment (mandziukGIT) [reply src/components/register/FinalStep.vue]: My aim was to just try it out but I will also move rest to setup 🔥

File: src/components/movies/MovieCard.vue (hunk 2 above)
Comment (mandziukGIT) [reply src/components/movies/MovieCard.vue]: I guess this time the only reason was my lack of knowledge. You are right 👌

**Review by szymonlicau — APPROVED** [review]
Nice usage of composition API!
Also nice to see some composables :heart:

**Pull request merged**